pkg/maven/maven.go

```diff
@@ -107,6 +134,146 @@ func Evaluate(options *EvaluateOptions, expression string, command mavenExecRunn
+	}
+	return nil
+}
+
+// InstallMavenArtifacts finds maven modules (identified by pom.xml files) and installs the artifacts into the local maven repository.
+func InstallMavenArtifacts(command mavenExecRunner, options EvaluateOptions) error {
+	return doInstallMavenArtifacts(command, options, newUtils())
+}
+
+func doInstallMavenArtifacts(command mavenExecRunner, options EvaluateOptions, utils mavenUtils) error {
+	err := flattenPom(command)
+	if err != nil {
+		return err
+	}
+
+	pomFiles, err := utils.glob(filepath.Join("**", "pom.xml"))
+	if err != nil {
+		return err
+	}
+
+	oldWorkingDirectory, err := utils.getwd()
+	if err != nil {
+		return err
+	}
+
+	for _, pomFile := range pomFiles {
+		log.Entry().Info("Installing maven artifacts from module: " + pomFile)
+		dir := path.Dir(pomFile)
+		err = utils.chdir(dir)
+		if err != nil {
+			return err
+		}
+
+		// Set pom path fix here because we cd'ed into the module's directory, so the pom is not in a subdirectory
+		options.PomPath = "pom.xml"
+		packaging, err := Evaluate(&options, "project.packaging", command)
+		if err != nil {
+			return err
+		}
+
+		if packaging == "pom" {
+			err = InstallFile("", "pom.xml", command)
+			if err != nil {
+				return err
+			}
+		} else {
+			err = installJarWarArtifacts(command, utils, options)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = utils.chdir(oldWorkingDirectory)
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func installJarWarArtifacts(command mavenExecRunner, utils mavenUtils, options EvaluateOptions) error {
+	// Set pom path fix here because we cd'ed into the module's directory, so the pom is not in a subdirectory
+	options.PomPath = "pom.xml"
+	finalName, err := Evaluate(&options, "project.build.finalName", command)
+	if err != nil {
+		return err
+	}
+	if finalName == "" {
+		log.Entry().Warn("project.build.finalName is empty, skipping install of artifact. Installing only the pom file.")
+		err = InstallFile("", "pom.xml", command)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	jarExists, err := utils.FileExists(jarFile(finalName))
+	if err != nil {
+		return err
+	}
+	warExists, err := utils.FileExists(warFile(finalName))
+	if err != nil {
+		return err
+	}
+
+	if jarExists {
+		err = InstallFile(jarFile(finalName), "pom.xml", command)
+		if err != nil {
+			return err
+		}
+	}
+
+	if warExists {
+		err = InstallFile(warFile(finalName), "pom.xml", command)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func jarFile(finalName string) string {
+	return "target/" + finalName + ".jar"
+}
+
+func warFile(finalName string) string {
+	return "target/" + finalName + ".war"
+}
+
+func flattenPom(command mavenExecRunner) error {
+	mavenOptionsFlatten := ExecuteOptions{
+		Goals:   []string{"flatten:flatten"},
+		Defines: []string{"-Dflatten.mode=resolveCiFriendliesOnly"},
+		PomPath: "pom.xml",
+	}
+	_, err := Execute(&mavenOptionsFlatten, command)
+	return err
+}
```

Comment:
Shouldn't this also be passed the M2Path, ProjectSettingsFile and GlobalSettingsFile?